plenum/client/wallet.py

```diff
@@ -265,3 +267,153 @@ def _getIdData(self,
         signer = self.idsToSigners.get(idr)
         idData = self.ids.get(idr)
         return IdData(signer, idData.lastReqId if idData else None)
+
+
+
+class WalletStorageHelper:
+    """Manages wallets
+
+    :param ``keyringsBaseDir``: keyrings base directory
+    :param dmode: (optional) permissions for directories inside
+        including the base one, default is 0700
+    :param fmode: (optional) permissions for files inside,
+        default is 0600
+    """
+    def __init__(self, keyringsBaseDir, dmode=0o700, fmode=0o600):
+        self.dmode = dmode
+        self.fmode = fmode
+        self.keyringsBaseDir = keyringsBaseDir
+
+
+    @property
+    def keyringsBaseDir(self):
+        return self._baseDir
+
+
+    @keyringsBaseDir.setter
+    def keyringsBaseDir(self, path):
+        self._baseDir = os.path.realpath(path)
+        self._baseDirParts = self._splitPath(self._baseDir)
+
+        self._createDirIfNotExists(self._baseDir)
+        self._ensurePermissions(self._baseDir, self.dmode)
+
+
+    def _ensurePermissions(self, path, mode):
+        if stat.S_IMODE(os.stat(path).st_mode) != mode:
+            os.chmod(path, mode)
+
+
+    def _createDirIfNotExists(self, dpath):
+        if os.path.exists(dpath):
+            if not os.path.isdir(dpath):
+                raise NotADirectoryError("{}".format(dpath))
+        else:
+            os.makedirs(dpath)
+
+
+    def _normalize(self, fpath):
+        # in case of relative path construct the absolute one
+        _fpath = os.path.realpath(fpath if os.path.isabs(fpath) else 
+                    os.path.join(self._baseDir, fpath))
+        parts = self._splitPath(_fpath)
+        return _fpath, parts[:-1], parts[-1]
+
+
+    def _splitByBaseDir(self, parts):
+        baseLen = len(self._baseDirParts)
+        return parts[:baseLen], parts[baseLen:]
+
+
+    def _splitPath(self, path):
```

Comment:
[Python docs](https://docs.python.org/2/library/os.html#os.sep) warns about doing that. At least on Windows we will get an issue:
```
C:\Users\and>python
Python 3.5.3 (v3.5.3:1880cb95a742, Jan 16 2017, 16:02:32) [MSC v.1900 64 bit (AM
D64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.sep
'\\'
>>> os.path.realpath('C:\\Users')
'C:\\Users'
>>> os.path.split(os.path.realpath('C:\\Users'))
('C:\\', 'Users')
>>> os.path.realpath('C:\\Users').split(os.sep)
['C:', 'Users']
```